include/InteractiveModelView.h

```diff
@@ -0,0 +1,159 @@
+/*
+ * InteractiveModelView.h - TODO
+ *
+ * Copyright (c) 2024 szeli1 <TODO/at/gmail/dot/com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef LMMS_GUI_INTERACTIVE_MODEL_VIEW_H
+#define LMMS_GUI_INTERACTIVE_MODEL_VIEW_H
+
+#include <list>
+#include <memory>
+#include <vector>
+
+#include <QApplication>
+#include <QWidget>
+#include <QColor>
+
+#include "Clipboard.h"
+#include "lmms_export.h"
+#include "ModelView.h"
+
+class QColor;
+class QMimeData;
+class QPainter;
+class QTimer;
+
+namespace lmms::gui
+{
+
+class SimpleTextFloat;
+
+class LMMS_EXPORT InteractiveModelView : public QWidget
+{
+Q_OBJECT
+	Q_PROPERTY(QColor highlightColor READ getHighlightColor WRITE setHighlightColor)
+public:
+	InteractiveModelView(QWidget* widget);
+	~InteractiveModelView() override;
+
+	static void startHighlighting(Clipboard::StringPairDataType dataType);
+	static void stopHighlighting();
+	static void showMessage(QString& message);
+	static void hideMessage();
+
+	static QColor getHighlightColor();
+	static void setHighlightColor(QColor& color);
+protected:
+	class ModelShortcut
+	{
+	public:
+		inline ModelShortcut() {}
```

Comment:
Constructors and other methods are already implicitly `inline` so the `inline` specifier can be removed.